### PR TITLE
Update NN to 87.0.0 and Maps to 10.3.0-beta.1

### PR DESCRIPTION
**Pull request by chezzdev**

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -24,8 +24,8 @@ let package = Package(
     dependencies: [
         .package(name: "MapboxDirections", url: "https://github.com/mapbox/mapbox-directions-swift.git", from: "2.2.0"),
         .package(name: "MapboxMobileEvents", url: "https://github.com/mapbox/mapbox-events-ios.git", from: "1.0.0"),
-        .package(name: "MapboxNavigationNative", url: "https://github.com/mapbox/mapbox-navigation-native-ios.git", from: "83.0.0"),
-        .package(name: "MapboxMaps", url: "https://github.com/mapbox/mapbox-maps-ios.git", from: "10.2.0"),
+        .package(name: "MapboxNavigationNative", url: "https://github.com/mapbox/mapbox-navigation-native-ios.git", from: "87.0.0"),
+        .package(name: "MapboxMaps", url: "https://github.com/mapbox/mapbox-maps-ios.git", from: "10.3.0-beta.1"),
```
Comment (MaximAlien) [inline Package.swift:28]: Looks like `MapboxMaps` has to be updated for CocoaPods based targets as well.

**Comment by 1ec5** [comment]
Upgrading to MapboxMaps v10.3.0-beta.1 is blocked by mapbox/mapbox-maps-ios#1073. We could either forego the MapboxMaps upgrade for now in favor of just upgrading to MapboxNavigationNative v87._x_, or we could treat the Xcode 12._x_ incompatibility as a known issue to be resolved in a subsequent prerelease of MapboxNavigation v2.3.0.

/cc @macdrevx

**Comment by 1ec5** [comment]
[Builds are still failing](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/7114/workflows/4b9f7f7f-0337-4233-8ba8-9cb87da6fa98/jobs/59183) on MapboxNavigationNative integration problems:

```
/Users/distiller/project/Sources/MapboxCoreNavigation/MapboxRoutingProvider.swift:108:36: error: value of type 'RouterInterfaceNative' (aka 'RouterInterface') has no member 'cancelRouteRefreshRequest'
            routingProvider.router.cancelRouteRefreshRequest(forToken: requestIdentifier)
            ~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~
/Users/distiller/project/Sources/MapboxCoreNavigation/MapboxRoutingProvider.swift:109:36: error: value of type 'RouterInterfaceNative' (aka 'RouterInterface') has no member 'cancelRouteRequest'
            routingProvider.router.cancelRouteRequest(forToken: requestIdentifier)
            ~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~~~~~~
/Users/distiller/project/Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:52:54: error: extra argument 'router' in call
                                             router: nil)
                                                     ^~~
/Users/distiller/project/Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:52:54: error: 'nil' requires a contextual type
                                             router: nil)
                                                     ^
/Users/distiller/project/Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:114:46: error: extra arguments at positions #6, #7 in call
        let navigatorConfig = NavigatorConfig(voiceInstructionThreshold: nil,
                                             ^
MapboxNavigationNative.NavigatorConfig:6:12: note: 'init(voiceInstructionThreshold:electronicHorizonOptions:polling:incidentsOptions:noSignalSimulationEnabled:)' declared here
    public init(voiceInstructionThreshold: NSNumber?, electronicHorizonOptions: ElectronicHorizonOptions?, polling: PollingConfig?, incidentsOptions: IncidentsOptions?, noSignalSimulationEnabled: NSNumber?)
           ^
/Users/distiller/project/Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:119:69: error: 'nil' requires a contextual type
                                              avoidManeuverSeconds: nil,
                                                                    ^
/Users/distiller/project/Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:120:59: error: 'nil' requires a contextual type
                                              useSensors: nil)
                                                          ^
```

**Comment by 1ec5** [comment]
These errors on CI are concerning. It’s as if the MapboxNavigationNative v87.0.1 binary has the same public API as v83.0.1, even though the source says otherwise. However, locally, I do see the correct public API (even if I can’t build it due to the map SDK incompatibility).

/cc @vanmaxim @SiarheiFedartsou

**Review by MaximAlien — APPROVED** [review]

**Pull request merged**